tasks/typescript/hard/main.ts

```diff
@@ -10,14 +10,30 @@ interface Product {
 let products: Product[] = [];
 
 async function fetchData() {
-    // TODO: Implement the fetch function
+    const response = await fetch('https://dummyjson.com/products');
+    const data = await response.json();
+    const products = data;
```

Comment:
Code looks mostly good. However, this code does not work as it is, since the endpoint returns an object and not array, so the `const products = data;` line would need to be changed to `products = data.products;` to get the actual products and also remove the const because we want to modify the global products.

After those changes it works, but this is close enough so i'm still going to merge it. In the future, remember to always test the code in your local machine before submitting a PR, otherwise we have to do some troubleshooting and that takes up time for no reason.